src/components/Subscriptions/SubscriptionsMonthlyGood.vue

```diff
@@ -248,17 +269,39 @@ export default {
 		result({ data }) {
 			this.isMonthlyGoodSubscriber = data?.my?.autoDeposit?.isSubscriber ?? false;
 			if (this.isMonthlyGoodSubscriber) {
-				const autoDepositAmount = parseFloat(data?.my?.autoDeposit?.amount ?? 0);
-				this.donation = parseFloat(data?.my?.autoDeposit?.donateAmount ?? 0);
+				const autoDepositAmount = parseFloat(data?.mySubscriptions?.values?.[0]?.amount ?? 0);
```

Comment:
I don't think these changes are correct. 

The legacy service used to have the total defined by autoDepositAmount and a donation. This is the reason, why we had to do this: `this.mgAmount = autoDepositAmount - this.donation;`

With the new subscription service, this is not necessary. 

This line can be removed: `const autoDepositAmount = parseFloat(data?.mySubscriptions?.values?.[0]?.amount ?? 0);`

and `this.mgAmount = autoDepositAmount - this.donation;` can now become: 

`parseFloat(data?.mySubscriptions?.values?.[0]?.amount ?? 0);`

Reply:
Also, before setting the component variables in this result block, you should check to see that this subscription is a monthly good subscription and not just use the first item: data?.mySubscriptions?.values?.[0]

I think this should be refactored to be: 
`const activeMonthlyGoodSubscription = data?.mySubscriptions?.values.find(sub => sub.subscriptionType === "MG" === "MG")` - similar to what you're doing to get the MG history values. 

Then set all your other variables based on that subscription

Reply:
Yeah that would work 👍